### ISSUE TITLE
Typescript warning/error

### ISSUE THREAD

**Issue by fpauser** [ember-redux/ember-rxjs-shim#1]
Created a fresh ember-addon via `ember addon rxjs-test` (ember-cli 2.14.2) and installed ember-rxjs-shim via `ember install ember-rxjs-shim`. Running `ember s` gives me:

```bash
Livereload server on http://localhost:49153
/src/observable/GenerateObservable.ts(182,9): Argument of type '(<T>(value: T) => T) | ResultFunc<S, T>' is not assignable to parameter of type 'ResultFunc<S, T>'.
  Type '<T>(value: T) => T' is not assignable to type 'ResultFunc<S, T>'.
    Type 'S' is not assignable to type 'T'.
/src/observable/GenerateObservable.ts(191,9): Argument of type '<T>(value: T) => T' is not assignable to parameter of type 'ResultFunc<S, T>'.
/src/observable/dom/AjaxObservable.ts(250,9): Type 'string' is not assignable to type 'XMLHttpRequestResponseType'.
```
I am just starting to play with typescript - any hints on this?

**Comment by fpauser** [comment]
Some more info regarding used typesript/rxjs versions:

```bash
$ yarn list | grep typescript
├─ broccoli-typescript-compiler@1.0.1
│  ├─ broccoli-typescript-compiler@^1.0.1
│  └─ typescript@^2.3.0
├─ typescript@2.4.2
```

```bash
$ yarn list | grep rxjs
├─ ember-rxjs-shim@1.0.0
├─ rxjs@5.4.3
```

**Comment by toranb** [comment]
Typescript 2.4  is likely at fault here. As a workaround - add typescript 2.3.4 to your app and you should see a big improvement. One struggle w/ TypeScript is that it's not semver so 2.4 is a breaking change for a lot of projects that worked fine using 2.3. 

![screen shot 2017-08-14 at 7 38 43 am](https://user-images.githubusercontent.com/147411/29271978-a4976a0e-80c3-11e7-9eed-e8113e9e7723.png)


Looking at the RxJS project itself I found they only use TypeScript ~2.0.6 to build it right now ... making me think 2.4 isn't actually supported at the moment.

https://github.com/ReactiveX/rxjs/blob/master/package.json#L197

It's possible we could/should alter this shim to explicitly use 2.3.4 instead of what I see today ^2.3.0

Thoughts? I assume making this change in the 1x series of the ember-rxjs-shim wouldn't harm anyone ... and it would make the compiler output less painful

**Comment by fpauser** [comment]
Sounds good to me. Only thing is that ember-cli-typescript (which I just added to my addon) also requires typescript ^2.4.2...

**Comment by toranb** [comment]
Ah, in that case I'd say the combination of Rx 5 and ember-cli-typescript won't be 100% error free. That said, I did look at the `next` branch for Rx just now and it appears they are making progress toward legit TypeScript 2.4+ support

https://github.com/ReactiveX/rxjs/commits/next

I'm not certain what the timeline is for Rx 6 but the moment it's out /stable we could update this shim to support it + ember-cli-typescript (including TypeScript v2.4)

**Comment by toranb** [comment]
@fpauser I decided to lock this shim down to TypeScript 2.3.0 explicitly 

Looking at RxJS itself they lock the current 5X series to TypeScript ~2.0.6 so supporting anything > 2.3.0 right now isn't a match IMHO and this was a bug (my fault). I've since updated that and published a v1.1.0 (error/ warning free)

Sorry it took me so long to pull the trigger on this update (I was fearful of a semver issue but in all honesty this just isn't how RxJS is expected to work so - a bug more than anything)
